apis/bigquerybiglake/v1alpha1/catalog_types.go

```diff
@@ -14,66 +14,86 @@
-// 	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
-
-// 	// A unique specifier for the BigLakeCatalog resource in GCP.
-// 	ExternalRef *string `json:"externalRef,omitempty"`
-
-// 	// ObservedState is the state of the resource as most recently observed in GCP.
-// 	ObservedState *BigLakeCatalogObservedState `json:"observedState,omitempty"`
-// }
-
-// // BigLakeCatalogObservedState is the state of the BigLakeCatalog resource as most recently observed in GCP.
-// // +kcc:proto=google.cloud.bigquery.biglake.v1.Catalog
-// type BigLakeCatalogObservedState struct {
-// }
-
-// // +genclient
-// // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-// // +kubebuilder:resource:categories=gcp,shortName=gcpbiglakecatalog;gcpbiglakecatalogs
-// // +kubebuilder:subresource:status
-// // +kubebuilder:metadata:labels="cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/system=true"
-// // +kubebuilder:printcolumn:name="Age",JSONPath=".metadata.creationTimestamp",type="date"
-// // +kubebuilder:printcolumn:name="Ready",JSONPath=".status.conditions[?(@.type=='Ready')].status",type="string",description="When 'True', the most recent reconcile of the resource succeeded"
-// // +kubebuilder:printcolumn:name="Status",JSONPath=".status.conditions[?(@.type=='Ready')].reason",type="string",description="The reason for the value in 'Ready'"
-// // +kubebuilder:printcolumn:name="Status Age",JSONPath=".status.conditions[?(@.type=='Ready')].lastTransitionTime",type="date",description="The last transition time for the value in 'Status'"
-
-// // BigLakeCatalog is the Schema for the BigLakeCatalog API
-// // +k8s:openapi-gen=true
-// type BigLakeCatalog struct {
-// 	metav1.TypeMeta   `json:",inline"`
-// 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-// 	// +required
-// 	Spec   BigLakeCatalogSpec   `json:"spec,omitempty"`
-// 	Status BigLakeCatalogStatus `json:"status,omitempty"`
-// }
-
-// // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-// // BigLakeCatalogList contains a list of BigLakeCatalog
-// type BigLakeCatalogList struct {
-// 	metav1.TypeMeta `json:",inline"`
-// 	metav1.ListMeta `json:"metadata,omitempty"`
-// 	Items           []BigLakeCatalog `json:"items"`
-// }
-
-// func init() {
-// 	SchemeBuilder.Register(&BigLakeCatalog{}, &BigLakeCatalogList{})
-// }
+// BigLakeCatalogSpec defines the desired state of BigLakeCatalog
+// +kcc:proto=google.cloud.bigquery.biglake.v1.Catalog
+type BigLakeCatalogSpec struct {
+	// The project that this resource belongs to.
+	// +required
+	ProjectRef *refs.ProjectRef `json:"projectRef,omitempty"`
+
+	// Immutable. The location where the Table should reside.
+	// +required
+	Location string `json:"location,omitempty"`
```

Comment:
Suggest moving project and location to a json inline parent.

Another question: Does this resource have no configurable field?

Reply:
no configurable field as of now: https://cloud.google.com/bigquery/docs/reference/biglake/rest/v1/projects.locations.catalogs